webots_ros2_core/webots_ros2_core/webots_launcher.py

```diff
@@ -30,9 +37,52 @@ def __init__(self, *, world, gui, mode):
         self.__mode = mode if isinstance(mode, Substitution) else TextSubstitution(text=mode)
         self.__world = world if isinstance(world, Substitution) else TextSubstitution(text=world)
 
+    def install_webots(self):
+        # Remove previous archive
+        installationDirectory = os.path.join(os.environ['HOME'], '.ros')
+        installationPath = os.path.abspath(os.path.join(installationDirectory, 'webots'))
+        archiveName = 'webots-%s-x86-64.tar.bz2' % get_required_webots_version_short()
+        archivePath = os.path.join(installationDirectory, archiveName)
+        if os.path.exists(archivePath):
+            os.remove(archivePath)
+        # Remove previous webots folder
+        if os.path.exists(installationPath):
+            shutil.rmtree(installationPath)
+        # Get Webots archive
+        print('\033[33mInstalling Webots (%s), this might take some time.\033[0m' % get_required_webots_version())
+        url = 'https://github.com/cyberbotics/webots/releases/download/%s/' % get_required_webots_version_short()
+        urllib.request.urlretrieve(url + archiveName, archivePath)
+        # Extract Webots archive
+        tar = tarfile.open(archivePath, 'r:bz2')
+        tar.extractall(os.path.join(installationDirectory, 'webots' + get_required_webots_version_short()))
+        tar.close()
+        os.remove(archivePath)
+        os.environ['WEBOTS_HOME'] = os.path.join(installationPath, 'webots')
+
     def perform(self, context):
-        # Add `webots` executable to command
         webots_path = get_webots_home()
+        if webots_path is None:
+            if context.perform_substitution(self.__gui).lower() in ['false', '0']:
+                sys.exit('Missing Webots version "%s"' % get_required_webots_version())
+            answer = messagebox.askyesno('Webots is missing.',
+                                         'Webots "%s" is missing, would you like to install it?' %
+                                         get_required_webots_version())
+            if answer:
+                self.install_webots()
+                webots_path = get_webots_home()
+                if webots_path is None:
+                    sys.exit('Failed to install Webots "%s"' % get_required_webots_version())
+            else:
+                answer = filedialog.askdirectory(
+                    initialdir=os.getcwd(),
+                    title="If Webots is already installed, please select its home installation folder:"
```

Comment:
It is not clear to me that if we answer no to the first dialog, we will have the possibility to specify the installation path in a second step. Maybe the first dialog should say something like "Webots R2020b revision 1 was not found in your system PATH." and have two buttons: "Install it" and "Specify path of already installed version" or something similar.

Reply:
> we will have the possibility to specify the installation path in a second step.

Yes, you have. I will improve the description of the first dialog to make this clearer.

Reply:
Changing the labels of the buttons is not so simple as we can't rely anymore on the default message box, but I have to create my own from scratch.